thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/DetectionJobResource.java

```diff
@@ -1190,4 +1191,30 @@ public Response getProjectedMTTD (@PathParam("autotuneId") @NotNull long id,
     double functionMTTDInHour = TimeUnit.HOURS.convert(detectionBucketSize, detectionUnit);
     return Response.ok(Math.max(functionMTTDInHour, alertFilterMTTDInHour)).build();
   }
+
+  /**
+   * Given autotuneId and a list of anomalies, return the anomaly ids that qualified for alert filter
+   * @param autotuneId
+   * @param anomalyIdsString comma split anomaly id string
+   * @return Array list of anomaly id that pass the alert filter in autotune
+   */
+  @GET
+  @Path("/eval/projected/anomalies/{autotuneId}")
+  public ArrayList<Long> getPreviewedAnomaliesByAutoTuneId (@PathParam("autotuneId") @NotNull long autotuneId,
```

Comment:
autotuneId won't be null because it is a long. Besides, it is part of the link.

Reply:
fixed. I see examples from other places have "Notnull" as PathParam. Those need to be refactored sometime